app/views/questionnaire.py

```diff
@@ -565,16 +566,24 @@ def perform_list_action(
         questionnaire_store_updater.add_list_item_and_answers(
             form, parent_block['populates_list']
         )
-    elif block['type'] == 'ListEditQuestion':
+    if block['type'] == 'ListEditQuestion':
         questionnaire_store_updater.update_answers(form)
 
+    if schema.is_primary_person_block_type(block['type']):
```

Comment:
I can't see why primary person uses the `perform_list_action` method. It would be clearer if this was called from `post_block_handler`.

Reply:
Done. Though annoyingly makes post_handler 'too complex'. I've ignored this for now on the basis that this is all going to be refactored fairly soon.